### PR TITLE
PEER-204 Endpoint for Registration

### PR DESCRIPTION
**Pull request by sembcorpp**
Managed to add the endpoints for user registrations:
Screenshot 2024-09-23 at 1 38 23 AM
Screenshot 2024-09-23 at 1 42 01 AM

### CODE REVIEW

File: backend/user/src/server.ts [hunk 1]
```diff
@@ -12,6 +13,7 @@ app.use(pino());
 app.use(json());
 app.use(helmet());
 app.use('/auth', authRoutes);
+app.use('/register', registerRoutes);
```
Comment (SeeuSim) [inline backend/user/src/server.ts]: @sembcorpp might wanna discuss with @TabrizPlv if this is the routing to go with, or to merge with auth routes

File: backend/user/src/services/registration/register-logic.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      lastName,
+      password: hashedPassword, //store the hashed password
+    })
+    .returning(_getSchema());
+
+  // return success response with the JWT token
+  return {
+    code: StatusCodes.CREATED,
+    data: {
+      user: newUser,
```
Comment (SeeuSim) [inline backend/user/src/services/registration/register-logic.ts:56]: Do not return the hashed password over REST

**Review by SeeuSim — CHANGES_REQUESTED** [review]

**Comment by sembcorpp** [comment]
removed the hashed passwords from return output and changed the endpoint for register to be under auth

Screenshot 2024-09-23 at 11 04 36 AM

**Review by SeeuSim — APPROVED** [review]
lgtm, please merge

**Pull request merged**